File: BaseBackend/src/main/java/com/common/model/sql/SystemSetting.java
package com.common.model.sql;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;

/**
 * Thực thể lưu trữ cấu hình, tham số hệ thống chạy động trong cơ sở dữ liệu SQL.<br/>
 * Kế thừa BaseModel để tự động quản lý các trường kiểm toán (Audit fields).<br/>
 * Created at 20/06/2026
 *
 * @see <a href="../../../../../resources/docs/spring_structure/system-settings-guide.md">System Settings Specification Guide</a>
 * @author txhoan
 */
@Entity
@Table(name = "system_settings")
@Getter
@Setter
@ToString(callSuper = true)
@SuperBuilder
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class SystemSetting extends BaseModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @Column(name = "setting_key", nullable = false, unique = true)
    String settingKey;

    @Column(name = "setting_value", nullable = false, length = 1000)
    String settingValue;

    @Column(name = "description")
    String description;
}

File: be/.Net/Swe.Common/Model/SystemSetting.cs
using System.ComponentModel.DataAnnotations;
using Swe.Common.Attributes;
using Swe.Common.Base;

namespace Swe.Common.Model;

/// <summary>
/// Thực thể lưu trữ cấu hình, tham số hệ thống chạy động trong cơ sở dữ liệu SQL.
/// </summary>
/// <created_at>2026-08-03</created_at>
/// <author>txhoan</author>
[ConfigTable("system_settings")]
public class SystemSetting : BaseModel
{
    [Key]
    [ConfigColumn("id")]
    public long Id { get; set; }

    [Required]
    [MaxLength(255)]
    [ConfigColumn("setting_key")]
    [CheckDuplicate("DuplicatedSettingKey")]
    public string SettingKey { get; set; } = string.Empty;

    [Required]
    [MaxLength(1000)]
    [ConfigColumn("setting_value")]
    public string SettingValue { get; set; } = string.Empty;

    [MaxLength(500)]
    [ConfigColumn("description")]
    public string? Description { get; set; }
}
